src/backend/joanie/core/factories.py

```diff
@@ -858,6 +859,18 @@ def billing_address(self, create, extracted, **kwargs):
 
                 self.init_flow(billing_address=BillingAddressDictFactory())
 
+        if (
+            not self.is_free
+            and self.has_contract
+            and target_state
+            not in [
+                enums.ORDER_STATE_DRAFT,
+                enums.ORDER_STATE_ASSIGNED,
+                enums.ORDER_STATE_TO_SIGN,
```

Comment:
I don't think, because  `generate_schedule()` is called in `order.submit_for_signature()`.
So only states after `ORDER_STATE_TO_SIGN` should generate a payment schedule.

Reply:
Yep you're right. I shuffled the order of signature and tokenization steps in my head.